src/SFML/Audio/AudioDevice.hpp

```diff
@@ -56,28 +58,16 @@ class AudioDevice
     ~AudioDevice();
 
     ////////////////////////////////////////////////////////////
-    /// \brief Check if an OpenAL extension is supported
-    ///
-    /// This functions automatically finds whether it
-    /// is an AL or ALC extension, and calls the corresponding
-    /// function.
-    ///
-    /// \param extension Name of the extension to test
-    ///
-    /// \return True if the extension is supported, false if not
-    ///
-    ////////////////////////////////////////////////////////////
-    static bool isExtensionSupported(const std::string& extension);
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Get the OpenAL format that matches the given number of channels
+    /// \brief Get the audio engine
     ///
-    /// \param channelCount Number of channels
+    /// There should only be a single instance of AudioDevice.
+    /// As long as an AudioResource exists, this function should
+    /// always return a valid pointer to the audio engine.
     ///
-    /// \return Corresponding format
+    /// \return The audio engine
     ///
     ////////////////////////////////////////////////////////////
-    static int getFormatFromChannelCount(unsigned int channelCount);
+    static void* getEngine();
```

Comment:
Why does this return a `void*` instead of a `ma_engine*`? I think it'd be cleaner and safer to instead forward declare `ma_engine` and use the type directly in the API.